Core/GUI/GameTooltipHooks.lua

```diff
@@ -510,6 +510,26 @@ local function processItem(id)
 	end
 end
 
+--- Handles updating an item tooltip to add additional information about attempts
+---@param tooltip GameTooltip
+---@param tooltipData TooltipData
+local function onTooltipSetItem(tooltip, tooltipData)
+	if tooltip ~= _G.GameTooltip and tooltip ~= _G.ItemRefTooltip then
+		return
+	end
+
+	local _, itemLink = tooltip:GetItem()
+	if not itemLink or type(itemLink) ~= "string" then
+		return
+	end
+
+	local id = itemLink:match("item:(%d+):")
```

Comment:
Any reason to worry about this match returning `nil`? Not sure if/how often the item link format has changed.

AFAICT `processItem` dodges that bullet, but maybe an extra `assert` could help avoid a future headache?

Reply:
Not sure, but does it hurt to have the check regardless? 🤔

Reply:
Technically yes since asserts aren't removed in Lua, but that's irrelevant here. I guess adding one would be great, then.

Reply:
I'll let you sort that out then as I've personally not dealt with asserts in Lua so I'd not want to mess it up!